source/utils/types.ts

```diff
@@ -77,6 +94,10 @@ export interface Options extends RequestOptions {
 	method?: Method;
 	retry?: number | RetryOption;
 	throwHttpErrors?: boolean;
+	cookieJar?: CookieJar;
+	request?: RequestFn;
+	agent: https.Agent | boolean | { [key: string]: https.Agent };
```

Comment:
What about `http.Agent`? Or is it exactly the same?

Instead of `[key: string]`, you can just explicitly list the keys. There are only two.

Reply:
> What about `http.Agent`? Or is it exactly the same?

`https.Agent` extends `http.Agent`. Its constructor takes a few additional options.

> Instead of `[key: string]`, you can just explicitly list the keys. There are only two.

Ah, sure, I'll update that. I wasn't sure if there were other protocols the library might support.